### PR TITLE
devcontainer: init at 0.58.0

### PR DESCRIPTION
**Pull request by Rucadi**
## Description of changes

Devcontainer (https://containers.dev/) is an opensource MIT implementation of the devcontainer specs. 

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Rucadi** [comment]
I had to move from mkYarnPackage -> stdenv.mkDerivation, it simplifies the derivation and doesn't make ofborg checks fail.

File: pkgs/by-name/de/devcontainer/package.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    mkdir -p $out/libexec
+    cp -r dist $out/libexec
+    cp devcontainer.js $out/libexec/devcontainer.js
+    cp -r node_modules $out/libexec/node_modules
+    runHook postBuild
+  '';
+
+  postInstall = ''
+    makeWrapper "${nodejs}/bin/node" "$out/bin/devcontainer" --add-flags "$out/libexec/devcontainer.js" \
+        --prefix PATH : ${lib.makeBinPath [ git docker docker-compose ]}
```
Comment (WolfangAukang) [inline pkgs/by-name/de/devcontainer/package.nix:45]: A nitpick
```suggestion
    makeWrapper "${nodejs}/bin/node" "$out/bin/devcontainer" \
      --add-flags "$out/libexec/devcontainer.js" \
      --prefix PATH : ${lib.makeBinPath [ git docker docker-compose ]}
```

File: pkgs/by-name/de/devcontainer/package.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
+{ lib, stdenv, fetchYarnDeps, fixup_yarn_lock, fetchFromGitHub, prefetch-yarn-deps, nodejs, python3, makeWrapper, git, docker, yarn, docker-compose }:
```
Comment (WolfangAukang) [inline pkgs/by-name/de/devcontainer/package.nix:1]:
```suggestion
{ lib
, stdenv
, fetchYarnDeps
, fixup_yarn_lock
, fetchFromGitHub
, prefetch-yarn-deps
, nodejs
, python3
, makeWrapper
, git
, docker
, yarn
, docker-compose
}:
```

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 301313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>devcontainer</li>
  </ul>
</details>

**Comment by Rucadi** [comment]
> Result of `nixpkgs-review pr 301313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package failed to build:
> ```
> last 25 log lines:
>        > source root is source
>        > Running phase: patchPhase
>        > Running phase: updateAutotoolsGnuConfigScriptsPhase
>        > Running phase: configurePhase
>        > no configure script, doing nothing
>        > Running phase: buildPhase
>        > yarn config v1.22.22
>        > success Set "yarn-offline-mirror" to "/nix/store/ylsg2883c01hh9j1c8myb2law2i5cwgq-offline".
>        > Done in 0.04s.
>        > yarn install v1.22.22
>        > [1/5] Validating package.json...
>        > [2/5] Resolving packages...
>        > [3/5] Fetching packages...
>        > [] 0/554[] 11/554[] 19/554[] 30/554[] 48/554[] 63/554[] 83/554[] 97/554[] 103/554[] 110/554[] 126/554[] 143/554[] 150/554[] 160/554[] 175/554[] 188/554[] 190/554[] 195/554[] 196/554[] 197/554[] 202/554[] 203/554[] 206/554[] 212/554[] 218/554[] 232/554[] 253/554[] 278/554[] 296/554[] 318/554[] 337/554[] 355/554[] 373/554[] 389/554[] 410/554[] 428/554[] 443/554[] 467/554[] 481/554[] 502/554[] 513/554[] 529/554[] 548/554[] 550/554[] 551/554[] 552/554[] 553/554[4/5] Linking dependencies...
>        > [] 554/554[] 0/589[] 0/16489[] 1021/16489[] 2155/16489[] 3290/16489[] 4432/16489[] 5693/16489[] 6960/16489[] 7938/16489[] 9203/16489[] 10098/16489[] 11445/16489[] 12960/16489[] 14471/16489[] 15738/16489[] 0/1017[5/5] Building fresh packages...
>        > 
>        > [] 1017/1017error /build/source/node_modules/esbuild: Command failed.
>        > Exit code: 127
>        > Command: node install.js
>        > Arguments:
>        > Directory: /build/source/node_modules/esbuild
>        > Output:
>        > /bin/sh: node: not found
>        > info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
>        > 
>        For full logs, run 'nix log /nix/store/hvmnhqj0s7a0n2aicfgdjp5khzx5lnww-devcontainer-0.58.0.drv'.
> error: 1 dependencies of derivation '/nix/store/mmp4x0x6jx43p60gwb4cyx7ncnvsvfxv-review-shell.drv' failed to build
> 
> Link to currently reviewing PR:
> https://github.com/NixOS/nixpkgs/pull/301313
> 
> 1 package failed to build:
> devcontainer
> ```

Wow! That was unexpected! Surprisingly I had no problems to build it..

> Result of `nixpkgs-review pr 301313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package failed to build:
> ```
> last 25 log lines:
>        > source root is source
>        > Running phase: patchPhase
>        > Running phase: updateAutotoolsGnuConfigScriptsPhase
>        > Running phase: configurePhase
>        > no configure script, doing nothing
>        > Running phase: buildPhase
>        > yarn config v1.22.22
>        > success Set "yarn-offline-mirror" to "/nix/store/ylsg2883c01hh9j1c8myb2law2i5cwgq-offline".
>        > Done in 0.04s.
>        > yarn install v1.22.22
>        > [1/5] Validating package.json...
>        > [2/5] Resolving packages...
>        > [3/5] Fetching packages...
>        > [] 0/554[] 11/554[] 19/554[] 30/554[] 48/554[] 63/554[] 83/554[] 97/554[] 103/554[] 110/554[] 126/554[] 143/554[] 150/554[] 160/554[] 175/554[] 188/554[] 190/554[] 195/554[] 196/554[] 197/554[] 202/554[] 203/554[] 206/554[] 212/554[] 218/554[] 232/554[] 253/554[] 278/554[] 296/554[] 318/554[] 337/554[] 355/554[] 373/554[] 389/554[] 410/554[] 428/554[] 443/554[] 467/554[] 481/554[] 502/554[] 513/554[] 529/554[] 548/554[] 550/554[] 551/554[] 552/554[] 553/554[4/5] Linking dependencies...
>        > [] 554/554[] 0/589[] 0/16489[] 1021/16489[] 2155/16489[] 3290/16489[] 4432/16489[] 5693/16489[] 6960/16489[] 7938/16489[] 9203/16489[] 10098/16489[] 11445/16489[] 12960/16489[] 14471/16489[] 15738/16489[] 0/1017[5/5] Building fresh packages...
>        > 
>        > [] 1017/1017error /build/source/node_modules/esbuild: Command failed.
>        > Exit code: 127
>        > Command: node install.js
>        > Arguments:
>        > Directory: /build/source/node_modules/esbuild
>        > Output:
>        > /bin/sh: node: not found
>        > info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
>        > 
>        For full logs, run 'nix log /nix/store/hvmnhqj0s7a0n2aicfgdjp5khzx5lnww-devcontainer-0.58.0.drv'.
> error: 1 dependencies of derivation '/nix/store/mmp4x0x6jx43p60gwb4cyx7ncnvsvfxv-review-shell.drv' failed to build
> 
> Link to currently reviewing PR:
> https://github.com/NixOS/nixpkgs/pull/301313
> 
> 1 package failed to build:
> devcontainer
> ```

This is totally unexpected since I can build the derivation myself, will try the nixpkgs review. 

Thanks a lot for the review!

**Comment by Rucadi** [comment]
> Result of `nixpkgs-review pr 301313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package built:
> * devcontainer

@WolfangAukang It seems that it is working fine for me... should I add nodejs to the nativeBuildInputs list or not?

File: pkgs/by-name/de/devcontainer/package.nix (hunk 2 above)
Comment (WolfangAukang) [reply pkgs/by-name/de/devcontainer/package.nix]: Honestly I had no idea of the RFC, so rather use it instead.

**Comment by WolfangAukang** [comment]
> should I add nodejs to the nativeBuildInputs list or not?

After seeing some examples on Nixpkgs, it doesn't look like it is necessary

File: pkgs/by-name/de/devcontainer/package.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+, prefetch-yarn-deps
+, nodejs
+, python3
+, makeWrapper
+, git
+, docker
+, yarn
+, docker-compose
+}:
+stdenv.mkDerivation rec {
```
Comment (WolfangAukang) [inline pkgs/by-name/de/devcontainer/package.nix:15]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/de/devcontainer/package.nix [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  meta = with lib; {
+    description = "Dev container CLI, run and manage your dev environments via a devcontainer.json";
+    homepage = "https://containers.dev/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rucadi ];
+    platforms = platforms.unix;
+    mainProgram = "devcontainer";
+  };
+}
```
Comment (WolfangAukang) [inline pkgs/by-name/de/devcontainer/package.nix:70]: To close the `finalAttrs` part
```suggestion
})
```

File: pkgs/by-name/de/devcontainer/package.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+, docker-compose
+}:
+stdenv.mkDerivation rec {
+  pname = "devcontainer";
+  version = "0.58.0";
+
+  src = fetchFromGitHub {
+    owner = "devcontainers";
+    repo = "cli";
+    rev = "v${version}";
```
Comment (WolfangAukang) [inline pkgs/by-name/de/devcontainer/package.nix:22]:
```suggestion
    rev = "v${finalAttrs.version}";
```

File: pkgs/by-name/de/devcontainer/package.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    hash = "sha256-pnhyyTJMSlTdMsSFzbmZ6SkGdbfr9qCIkrBxxSM42UE=";
+  };
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
```
Comment (WolfangAukang) [inline pkgs/by-name/de/devcontainer/package.nix:27]:
```suggestion
    yarnLock = finalAttrs.src + "/yarn.lock";
```

File: pkgs/by-name/de/devcontainer/package.nix [hunk 7]
```diff
@@ -0,0 +1,70 @@
+{ lib
+, stdenv
+, fetchYarnDeps
+, fixup_yarn_lock
```
Comment (WolfangAukang) [inline pkgs/by-name/de/devcontainer/package.nix:4]: :ou are picking the `fixup-yarn-lock` from somewhere else, so this is not needed.
```suggestion
```

File: pkgs/by-name/de/devcontainer/package.nix [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    hash = "sha256-Wy0UP8QaQzZ1par7W5UhnRLc5DF2PAif0JIZJtRokBk=";
+  };
+
+  nativeBuildInputs = [ yarn prefetch-yarn-deps python3 makeWrapper ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    export HOME=$(mktemp -d)
+    yarn config --offline set yarn-offline-mirror ${yarnOfflineCache}
```
Comment (WolfangAukang) [inline pkgs/by-name/de/devcontainer/package.nix:37]:
```suggestion
    yarn config --offline set yarn-offline-mirror ${finalAttrs.yarnOfflineCache}
```

**Comment by Rucadi** [comment]
@WolfangAukang could you approve the PR if it is ok for you?

**Comment by bbigras** [comment]
Result of `nixpkgs-review pr 301313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>devcontainer</li>
  </ul>
</details>

**Comment by bbigras** [comment]
@Rucadi are you able to use `devcontainer` with https://github.com/microsoft/vscode-remote-try-rust ? I also tried with the Go one.

```sh
❯ git clone https://github.com/microsoft/vscode-remote-try-rust
❯ /home/bbigras/.cache/nixpkgs-review/pr-301313/results/devcontainer/bin/devcontainer up --workspace-folder ./vscode-remote-try-rust
[1 ms] @devcontainers/cli 0.58.0. Node.js v20.11.1. linux 6.8.4 x64.
Error: ENOENT: no such file or directory, open '/nix/store/vbpyr6xdfrzsirsj2745d0z21yyva2kl-devcontainer-0.58.0/libexec/scripts/updateUID.Dockerfile'
{"outcome":"error","message":"ENOENT: no such file or directory, open '/nix/store/vbpyr6xdfrzsirsj2745d0z21yyva2kl-devcontainer-0.58.0/libexec/scripts/updateUID.Dockerfile'","description":"An error occurred setting up the container."}
```

**Comment by Rucadi** [comment]
Let me take a look, We are using this derivation at work successfully.

**Comment by Rucadi** [comment]
@bbigras can you try this version.
This is an older version that I did that for some reason didn't pass all the checks, and the errors seemed unrelated to my code and I decided to refactor it.

If this works, maybe you can help me fix the checks issues.

**Comment by bbigras** [comment]
```
❯ LANG=c nix run github:Mic92/nixpkgs-review -- pr 301313 --post-result
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/301313/head:refs/nixpkgs-review/1
$ git worktree add /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs f173d0881eff3b21ebb29a2ef8bedbc106c86ea5
Preparing worktree (detached HEAD f173d0881eff)
Updating files: 100% (40890/40890), done.
HEAD is now at f173d0881eff Merge pull request #303351 from asymmetric/mnt-pkgs
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs nixpkgs-overlays=/tmp/tmpmm7e2izr -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 115c17de80d18c081b9699b36fe941173f3857e7
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs nixpkgs-overlays=/tmp/tmpmm7e2izr -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
error:
       … while evaluating the 'name' attribute of a derivation

         at «none»:0: (source not available)

       … from call site

         at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:349:9:

          348|         in
          349|         lib.strings.sanitizeDerivationName (
             |         ^
          350|           if attrs ? name

       … while calling anonymous lambda

         at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/lib/strings.nix:1192:3:

         1191|   in
         1192|   string:
             |   ^
         1193|   # First detect the common case of already valid strings, to speed those up

       … from call site

         at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:273:18:

          272|       version = attrs.version or package.version;
          273|       baseName = unlessNull name "${safeName}-${version}";
             |                  ^
          274|

       … while calling 'unlessNull'

         at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:28:22:

           27|
           28|   unlessNull = item: alt:
             |                      ^
           29|     if item == null then alt else item;

       … from call site

         at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:271:18:

          270|       pname = attrs.pname or package.name;
          271|       safeName = reformatPackageName package.name;
             |                  ^
          272|       version = attrs.version or package.version;

       … while calling 'reformatPackageName'

         at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:31:25:

           30|
           31|   reformatPackageName = pname:
             |                         ^
           32|     let

       … from call site

         at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:269:17:

          268|     let
          269|       package = lib.importJSON packageJSON;
             |                 ^
          270|       pname = attrs.pname or package.name;

       … while calling 'importJSON'

         at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/lib/trivial.nix:639:16:

          638|   */
          639|   importJSON = path:
             |                ^
          640|     builtins.fromJSON (builtins.readFile path);

       error: cannot build '/nix/store/6xrj98a0bbdij8j206zn2hy4lp3hvy96-source.drv^out' during evaluation because the option 'allow-import-from-derivation' is disabled
https://github.com/NixOS/nixpkgs/pull/301313 failed to build: Failed to list packages: nix-env failed with exit code 1
$ git worktree remove -f /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs
```

**Comment by Rucadi** [comment]
> ```
> ❯ LANG=c nix run github:Mic92/nixpkgs-review -- pr 301313 --post-result
> $ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/301313/head:refs/nixpkgs-review/1
> $ git worktree add /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs f173d0881eff3b21ebb29a2ef8bedbc106c86ea5
> Preparing worktree (detached HEAD f173d0881eff)
> Updating files: 100% (40890/40890), done.
> HEAD is now at f173d0881eff Merge pull request #303351 from asymmetric/mnt-pkgs
> $ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs nixpkgs-overlays=/tmp/tmpmm7e2izr -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
> $ git merge --no-commit --no-ff 115c17de80d18c081b9699b36fe941173f3857e7
> Automatic merge went well; stopped before committing as requested
> $ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs nixpkgs-overlays=/tmp/tmpmm7e2izr -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
> error:
>        … while evaluating the 'name' attribute of a derivation
> 
>          at «none»:0: (source not available)
> 
>        … from call site
> 
>          at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:349:9:
> 
>           348|         in
>           349|         lib.strings.sanitizeDerivationName (
>              |         ^
>           350|           if attrs ? name
> 
>        … while calling anonymous lambda
> 
>          at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/lib/strings.nix:1192:3:
> 
>          1191|   in
>          1192|   string:
>              |   ^
>          1193|   # First detect the common case of already valid strings, to speed those up
> 
>        … from call site
> 
>          at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:273:18:
> 
>           272|       version = attrs.version or package.version;
>           273|       baseName = unlessNull name "${safeName}-${version}";
>              |                  ^
>           274|
> 
>        … while calling 'unlessNull'
> 
>          at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:28:22:
> 
>            27|
>            28|   unlessNull = item: alt:
>              |                      ^
>            29|     if item == null then alt else item;
> 
>        … from call site
> 
>          at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:271:18:
> 
>           270|       pname = attrs.pname or package.name;
>           271|       safeName = reformatPackageName package.name;
>              |                  ^
>           272|       version = attrs.version or package.version;
> 
>        … while calling 'reformatPackageName'
> 
>          at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:31:25:
> 
>            30|
>            31|   reformatPackageName = pname:
>              |                         ^
>            32|     let
> 
>        … from call site
> 
>          at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:269:17:
> 
>           268|     let
>           269|       package = lib.importJSON packageJSON;
>              |                 ^
>           270|       pname = attrs.pname or package.name;
> 
>        … while calling 'importJSON'
> 
>          at /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs/lib/trivial.nix:639:16:
> 
>           638|   */
>           639|   importJSON = path:
>              |                ^
>           640|     builtins.fromJSON (builtins.readFile path);
> 
>        error: cannot build '/nix/store/6xrj98a0bbdij8j206zn2hy4lp3hvy96-source.drv^out' during evaluation because the option 'allow-import-from-derivation' is disabled
> https://github.com/NixOS/nixpkgs/pull/301313 failed to build: Failed to list packages: nix-env failed with exit code 1
> $ git worktree remove -f /home/bbigras/.cache/nixpkgs-review/pr-301313-5/nixpkgs
> ```

That's the issue I don't really know how to solve with this version because this seems to come from yarn2nix :\

Another thing we could try is to put the package as a nodePackage here? pkgs/development/node-packages 

I have not tested it.

**Comment by Rucadi** [comment]
@bbigras  Copying the scripts folder to libexec worked, got go container running, can you test it?
```
rucadi@Rucadi:~/vscode-remote-try-go$ ./result/bin/devcontainer up  --workspace-folder .
[1 ms] @devcontainers/cli 0.58.0. Node.js v20.11.1. linux 5.15.146.1-microsoft-standard-WSL2 x64.
[60684 ms] Start: Run: docker run --sig-proxy=false -a STDOUT -a STDERR --mount type=bind,source=/home/rucadi/vscode-remote-try-go,target=/workspaces/vscode-remote-try-go -l devcontainer.local_folder=/home/rucadi/vscode-remote-try-go -l devcontainer.config_file=/home/rucadi/vscode-remote-try-go/.devcontainer/devcontainer.json --init --cap-add SYS_PTRACE --security-opt seccomp=unconfined --entrypoint /bin/sh -l devcontainer.metadata=[{"id":"ghcr.io/devcontainers/features/common-utils:2"},{"id":"ghcr.io/devcontainers/features/git:1"},{"id":"ghcr.io/devcontainers/features/go:1","init":true,"capAdd":["SYS_PTRACE"],"securityOpt":["seccomp=unconfined"],"customizations":{"vscode":{"extensions":["golang.Go"]}}},{"id":"ghcr.io/devcontainers/features/node:1","customizations":{"vscode":{"extensions":["dbaeumer.vscode-eslint"]}}},{"customizations":{"vscode":{"settings":{"go.toolsManagement.checkForUpdates":"local","go.useLanguageServer":true,"go.gopath":"/go"},"extensions":["golang.Go"]}},"remoteUser":"vscode"},{"customizations":{"vscode":{"settings":{},"extensions":["streetsidesoftware.code-spell-checker"]}},"portsAttributes":{"9000":{"label":"Hello Remote World","onAutoForward":"notify"}}}] vsc-vscode-remote-try-go-1e8f635eb937962d232785b5335359d3eaea1fb45269cf7c4ae24bbfb145c935-uid -c echo Container started
Container started
{"outcome":"success","containerId":"f9f5c39363028ec688bc3f48c085650ee2a7c7419c2cdaaea7e9c2fe5b5a5f46","remoteUser":"vscode","remoteWorkspaceFolder":"/workspaces/vscode-remote-try-go"}
rucadi@Rucadi:~/vscode-remote-try-go$ 
```

**Comment by bbigras** [comment]
```
❯ /home/bbigras/.cache/nixpkgs-review/pr-301313-7/results/devcontainer/bin/devcontainer up --workspace-folder .
[1 ms] @devcontainers/cli 0.58.0. Node.js v20.11.1. linux 6.8.4 x64.
[339 ms] Start: Run: docker start 03afc9bc34e32e6e05840bf1910f46695e141633fc54c3c961033c0c29752b81
[475 ms] 03afc9bc34e32e6e05840bf1910f46695e141633fc54c3c961033c0c29752b81
{"outcome":"success","containerId":"03afc9bc34e32e6e05840bf1910f46695e141633fc54c3c961033c0c29752b81","remoteUser":"vscode","remoteWorkspaceFolder":"/workspaces/vscode-remote-try-rust"}
```

:+1:

**Review by bbigras — APPROVED** [review]

**Review by sikmir — APPROVED** [review]

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 301313` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>devcontainer</li>
  </ul>
</details>

**Pull request merged**